bears/documentation/DocumentationStyleBear.py

```diff
@@ -0,0 +1,118 @@
+                following lines are also 4 space indented
+
+        :param language: The programming language of the file(s).
+        :param docstyle: The docstyle to use. For example ``default`` or
+                         ``doxygen``. Docstyles are language dependent, meaning
+                         not every language is supported by a certain docstyle.
+        :param allow_missing_func_desc: When set ``True`` this will allow
+                         functions with missing descriptions, allowing
+                         functions to start with params.
+        """
+        for doc_comment in extract_documentation(file, language, docstyle):
+            parsed = doc_comment.parse()
+            metadata = iter(parsed)
+
+            # Assuming that the first element is always the only main
+            # description.
+            main_description = next(metadata)
+
+            if main_description.desc == '\n' and not allow_missing_func_desc:
+                warning_desc = """
+Missing function description.
+Please set allow_missing_func_desc = True to ignore this warning.
+"""
+            else:
+                warning_desc = 'Documentation does not have correct style.'
+
+            # one empty line shall follow main description (except it's empty
+            # or no annotations follow).
+            if main_description.desc.strip() != '':
+                main_description = main_description._replace(
+                    desc='\n' + main_description.desc.strip() + '\n' *
+                         (1 if len(parsed) == 1 else 2))
+
+            new_metadata = [main_description]
+            for m in metadata:
+                # Split newlines and remove leading and trailing whitespaces.
+                stripped_desc = list(map(str.strip, m.desc.splitlines()))
+
+                if len(stripped_desc) == 0:
+                    # Annotations should be on their own line, though no
+                    # further description follows.
+                    stripped_desc.append('')
+                else:
+                    # Wrap parameter description onto next line if it follows
+                    # annotation directly.
+                    if stripped_desc[0] != '':
+                        stripped_desc.insert(0, '')
+
+                # Indent with 4 spaces.
+                stripped_desc = ('' if line == '' else ' ' * 4 + line for line
+                                 in stripped_desc)
+
+                new_desc = '\n'.join(stripped_desc)
+
+                # Strip away trailing whitespaces and obsolete newlines (except
+                # one newline which is mandatory).
+                new_desc = new_desc.rstrip() + '\n'
+
+                new_metadata.append(m._replace(desc=new_desc.lstrip(' ')))
+
+            new_comment = DocumentationComment.from_metadata(
+                new_metadata, doc_comment.docstyle_definition,
+                doc_comment.marker, doc_comment.indent, doc_comment.range)
```

Comment:
`doc_comment.indent` appears to be *guessed* at https://github.com/coala/coala/blob/master/coalib/bearlib/languages/documentation/DocumentationExtraction.py#L190 .

Again, this really should be determine by the user setting `indent_size` , if the user specified it , which would need to be passed into the `extract_documentation()` invocation.  But it would be a very low priority bug, as guessing is probably going to work well in most cases.  Could you take a quick look though this, confirm my suspicions, and raise a bug in coala/coala if I am correct.  It doesnt need to be solved for this bear, but we do need to know the potential bugs.